src/cpp/rtps/history/TopicPayloadPoolRegistry_impl/TopicPayloadPoolRegistry.hpp

```diff
@@ -37,12 +37,10 @@ class TopicPayloadPoolRegistry
 
 public:
 
-    ~TopicPayloadPoolRegistry() = default;
-
     /// @return reference to singleton instance
-    static TopicPayloadPoolRegistry& instance()
+    static const std::shared_ptr<TopicPayloadPoolRegistry>& instance()
     {
-        static TopicPayloadPoolRegistry pool_registry_instance;
+        static auto pool_registry_instance = std::shared_ptr<TopicPayloadPoolRegistry>(new TopicPayloadPoolRegistry());
```

Comment:
```suggestion
        static auto pool_registry_instance = std::make_shared<TopicPayloadPoolRegistry>();
```

Reply:
`make_shared` requires a public constructor, which I didn't want to provide in a singleton. The constructor of `shared_ptr` can access the private constructor if the context allows it. Same for the rest of classes

Reply:
You could then do the following:
```suggestion
        static std::shared_ptr<TopicPayloadPoolRegistry> pool_registry_instance{new TopicPayloadPoolRegistry()};
```